midas/input_parser.py

```diff
@@ -55,13 +55,13 @@ def validate_input(keyword, value):
     
     elif keyword == 'optimizer':
         value = str(value).lower().replace(' ','_')
-        if value not in ["genetic_algorithm"]:
-            raise ValueError("Requested optimizer name is invalid or not supported.")
+        if value not in ["genetic_algorithm","bayesian_optimization"]:
+            raise ValueError("Requested methodology '" + self.methodology + "' invalid.")
```

Comment:
"self" is not passed to validate_input(), also, self.methodology has not been defined yet. The value you're wanting to used is stored in "value".